packages/core/src/services/ai/index.ts

```diff
@@ -90,6 +92,67 @@ export async function ai({
     tools: toolsResult.value,
   }
 
+  if (UNSUPPORTED_STREAM_MODELS.includes(model)) {
+
+    if (output && schema) {
+      const result = await generateObject({
+        ...commonOptions,
+        schema: jsonSchema(schema),
+        output: output as any,
+      })
+
+      const fullStream = new ReadableStream<StreamChunk>({
+        start(controller: any) {
+          const streamChunk: StreamChunk = {
+            type: 'object',
+            object: result.object,
+          }
+          controller.enqueue(streamChunk)
+          controller.close()
+        },
+      })
+
+      return Result.ok({
+        type: 'object',
+        data: {
+          fullStream: fullStream as any,
+          object: Promise.resolve(result.object),
+          usage: Promise.resolve(result.usage),
+        },
+      })
+    }
+
+    const result = await generateText(commonOptions)
+
+    const fullStream = new ReadableStream<StreamChunk>({
+      start(controller: any) {
+        const streamChunk: StreamChunk = {
+          type: 'text-delta',
+          textDelta: result.text,
+        }
+        controller.enqueue(streamChunk)
+        controller.close()
+      },
+    })
+
+    const toolCalls = result.toolCalls.map((toolCall) => ({
+      type: 'tool-call' as const,
+      toolCallId: toolCall.toolCallId,
+      toolName: toolCall.toolName,
+      args: toolCall.args,
+    }))
+
+    return Result.ok({
+      type: 'text',
+      data: {
+        fullStream: fullStream as any,
+        text: Promise.resolve(result.text),
+        usage: Promise.resolve(result.usage),
+        toolCalls: Promise.resolve(toolCalls),
+      },
+    })
+  }
```

Comment:
If you could add some unit tests for this in the accompanying test file it would be awesome 🙏🏼

Reply:
Done :)